modules/gmc-agent/src/main/java/org/gridgain/agent/Agent.java

```diff
@@ -364,9 +364,18 @@ private void writeToMetaStorage(ManagementConfiguration cfg) {
                 if (disconnected.compareAndSet(false, true)) {
                     log.error("Lost websocket connection with server: " + curSrvUri);
 
-                    execSrvc.submit(Agent.this::connect0);
+                    submitConnectTask();
                 }
             }
         }
     }
+
+    /**
+     * Send the connect task to executor service.
+     */
+    private void submitConnectTask() {
+        // Submit a reconnection task only if we are no longer trying to reconnect
```

Comment:
Lets move this comment to javadocs of this method.

Reply:
Done.